service/src/main_impl.rs

```diff
@@ -815,7 +815,7 @@ where
 	});
 
 	// we attempt to set shard creation for this parentchain in case it hasn't been done before
-	let api_head = node_api.get_header(None).unwrap().unwrap();
+	let api_head = node_api.get_header(node_api.get_finalized_head().unwrap()).unwrap().unwrap();
```

Comment:
with this, the ancestry mismatch is gone.

I think it's acceptable, but we could also solve this by delaying parentchain initial sync until the creation header has been finalized

Reply:
Hmm, this is interesting. If this was the cause, it implies that fast-sync has never worked in the first place and that syncing from scratch is broken since January 2024 if we can trust git blame: https://github.com/integritee-network/worker/pull/1573

Can this really be true?

Reply:
Not necessarily:  the error only appears if the TargetA sync happens too fast after shard creation. I could imagine that we had to restart the worker a couple of times when we launched on Paseo.

But it could actually be that we used snapshots of light client state which predated Jan 2024. S, we don't really know

I guess I tested it with two solochains and not a zombienet setup which changes the finality timing